org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractMultiQuickfixTest.java

```diff
@@ -65,6 +65,29 @@ public static void afterClass() throws Exception {
 		IResourcesSetupUtil.setAutobuild(WAS_AUTOBUILD);
 	}
 
+	/**
+	 * Test that the multi-quickfixes work as expected.
+	 * 
+	 * @param initialText
```

Comment:
wonder why there need to be two initalText params

Reply:
The first parameter is for the DSL text, while the second one is for the DSL text enhanced by the recognized marker information. Such a call invocation could look like for example:

```xtend
	@Test def test001() {
		'''
			package foo
			class Foo {
				def m() {
					val a = new Object
					val b = new Object
					
					if (a == null) {}
					if (b != null) {}
				}
			}
		'''.testMultiQuickfix('''
			package foo
			class Foo {
				def m() {
					val a = new Object
					val b = new Object
					
					if (a <0<==>0> null) {}
					if (b <1<!=>1> null) {}
				}
			}
			-------------------------------
			0: message=The operator '==' should be replaced by '===' when null is one of the arguments.
			1: message=The operator '!=' should be replaced by '!==' when null is one of the arguments.
		''', '''
			package foo
			class Foo {
				def m() {
					val a = new Object
					val b = new Object
					
					if (a === null) {}
					if (b !== null) {}
				}
			}
			--------------------------
			(no markers found)
		''')
	}
```

Reply:
but how is the marker information extracted from then? isnt it sufficient to have the second parameter only?

Reply:
If you have only the second parameter, than the additional marker information has to be removed from it to get the initial DSL text. That is really difficult in case of complex dsls, I think.

Reply:
i still dont like this. at other places we remove markers too

Reply:
Is the `--------------------------` separator a static constant, or variable? I am getting...

    expected:<...--------------------[]
    0: message=The oper...> but was:<...--------------------[---------------------------------------]
    0: message=The oper...>

...when [I attempted to consume this API](https://github.com/drkstr101/xtext-xtend/blob/43594bae47cde2a13fc4a3f8a289b26fbb796b20/org.eclipse.xtend.ide.tests/longrunning/src/org/eclipse/xtend/ide/tests/quickfix/EqualsWithNullMultiQuickfixTest.xtend#L66).

Reply:
It is dynamically calculated and have the same length as the longest line in the dsl.

Reply:
What are everyone's thoughts on an API that works more like...?

```
val initialText = '''
    package foo
    class Foo {
        def m(Object a, Object b, Object c) {
            if(a == null || b != null || c === null) 0 else 1
        }
    }
'''
val expectedText = '''
    package foo
    class Foo {
        def m(Object a, Object b, Object c) {
            if(a == null || b != null || c === null) 0 else 1
        }
    }
'''
val markerText = '''
    0: message=The operator '==' should be replaced by '===' when null is one of the arguments.
    1: message=The operator '!=' should be replaced by '!==' when null is one of the arguments.
'''


testMultiQuickfix(initialText, expectedText, markerText)
```

Perhaps making the `markerText` value optional.

Reply:
Good idea! I was also thinking about providing more overloaded versions of the `testMultiQuickfix` method.
@cdietrich @szarnekow @kthoms @ArneDeutsch What do you think?